### PR TITLE
SqlDatabaseChain with Postgres implementation 

### PR DESCRIPTION
**Pull request by khoroshevj**
https://github.com/tryAGI/LangChain/issues/73
https://github.com/tryAGI/LangChain/issues/75

there is no [SQLAlchemy](https://www.sqlalchemy.org/) like lib for .net, so I added SqlDatabase base class and for each db type separate implementation needed.

added LangChain.Utilities.PostgresDatabases.IntegrationTests with SqlDatabaseChain test using Postgres+OpenAI

### CODE REVIEW

**Comment by HavenDV** [comment]
Very cool code quality.
Maybe we can simplify the project names a little (as a result, the names of NuGet packages) - instead of LangChain.Utilities.PostgresDatabases just LangChain.Utilities.Postgres and the same for SqlDatabases?

File: src/libs/Utilities/LangChain.Utilities.SqlDatabases/LangChain.Utilities.SqlDatabases.csproj [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>net4.6.2;netstandard2.0;net6.0;net7.0;net8.0</TargetFrameworks>
+        <NoWarn>$(NoWarn);CA1031</NoWarn>
+    </PropertyGroup>
+
+    <PropertyGroup Label="NuGet">
+        <Description>LangChain Utilities SqlDatabase.</Description>
+        <PackageTags>$(PackageTags);utilities;sqldatabase</PackageTags>
```
Comment (HavenDV) [inline src/libs/Utilities/LangChain.Utilities.SqlDatabases/LangChain.Utilities.SqlDatabases.csproj:10]: utilities PackageTag already specified in Directory.Build.props

File: src/libs/Utilities/LangChain.Utilities.PostgresDatabases/LangChain.Utilities.PostgresDatabases.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>net4.6.2;netstandard2.0;net6.0;net7.0;net8.0</TargetFrameworks>
+        <NoWarn>$(NoWarn);CA1031</NoWarn>
+    </PropertyGroup>
+
+    <PropertyGroup Label="NuGet">
+        <Description>LangChain Utilities PostgresDatabase.</Description>
+        <PackageTags>$(PackageTags);utilities;sqldatabase;postgres</PackageTags>
```
Comment (HavenDV) [inline src/libs/Utilities/LangChain.Utilities.PostgresDatabases/LangChain.Utilities.PostgresDatabases.csproj:10]: utilities PackageTag already specified in Directory.Build.props

**Comment by khoroshevj** [comment]
changed project names and added few more tests.
btw, it is not fully implemented, I'll add a comment to https://github.com/tryAGI/LangChain/issues/75 so we can merge this and implement rest later

**Comment by khoroshevj** [comment]
@HavenDV comments resolved, everything else fine?

**Pull request merged**